Move all WordPressAuthenticator sources into this repo

mokagio opened the pull request:
> [!NOTE]
> Closed in favor of https://github.com/woocommerce/woocommerce-ios/pull/15000 which preserves the Git history of the libraries folder into the project.

## Description

Removes the [`WordPressAuthenticator-iOS`](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS) and [`WordPressKit-iOS`](https://github.com/wordpress-mobile/WordPressKit-iOS) dependencies by _folding_ both into the codebase. See rationale in https://github.com/woocommerce/woocommerce-ios/issues/14329.

I implemented this using one framework target in the WooCommerce project, which is then  linked in the WooCommerce iOS app. I had initially tried having the code in a dedicated project, like the other frameworks like Yosemite, Storage, Networking, etc. do, but I run into some compilation issues and decided to move to the target setup. For what is worth, that is the same approach that WordPress and Jetpack use.

We could consider extracting into a project in the future, once most of the unused code (and presumably some additional dependency) has been removed.

Alternatively, we could move the whole code into the WooCommerce target itself. However I wouldn't recommend this because that target is already quite big and with tests that take a long time to run.

There is of course _a lot_ of new code in this PR. As you can see from the commit history, I spent some time deleting unused code, but I'm nowhere near done. I decided to draw a line in the sand and ask for review now, though, and continue chipping away at the unused afterwards. Otherwise, this will become stale and we'll risk more work happening in the libraries.

<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
N.A.

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
If CI is green and we can log into the app, then I'd call the operation successful.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
N.A.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. – N.A.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

mokagio committed:
Move all WordPressAuthenticator sources into this repo

mokagio committed:
Move iOS deployment target definition to workspace level

mokagio committed:
Update Xcode projects to 16.2 recommended settings

mokagio committed:
Fix location of shared xcconfig in Woo project

mokagio committed:
Add WordPressAuthenticator to WooCommerce workspace

mokagio committed:
Update `.editorconfig` to set explicit Swift settings

mokagio committed:
Update WordPressAuthenticator Objective-C headers access control

mokagio committed:
Link WordPressAuthenticator via local code instead of CocoaPods

mokagio committed:
Delete no longer necessary `WPAuthenticator-Swift.h` hack

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Disable module verifier for UITestsFoundation target

It got turned on by the 16.2 update and resulted in build failures.

mokagio committed:
Disable user script sandboxing for those tests targets that failed

mokagio committed:
Temporarily import WordPressKit via CP in main target

mokagio committed:
Temporarily import NSURL+IDN via CP in main target

mokagio committed:
Temporarily import SVProgressHUD via CP in main target

mokagio committed:
Rename folders in WordPressAuthenticator

mokagio committed:
Disabled SwiftLint `line_length` and address other violations

mokagio committed:
Remove an unnecessary import

mokagio committed:
Fold NSURL+IDN into the project — Remove CP dependency

mokagio committed:
Add flag for fetching code signing in `build_for_*` lanes

mokagio committed:
Update release toolkit to latest verison 12.4.0

mokagio committed:
Avoid archive error when building without fetching code signing

mokagio committed:
Create new, empty, WordPressKit

mokagio committed:
Use `Sources` and `Tests`

mokagio committed:
Set iOS deployment target to 16.0

mokagio committed:
Make WordPressAuthenticator a target in WooCommerce

Works better with the nested dependencies and the archiving checks.

mokagio committed:
Update Buildkite test collector version

mokagio committed:
Get WordPressAuthenticator to build with WordPressKit sources

mokagio committed:
Get WordPressAuthenticator tests to build and pass with WordPressKit

mokagio committed:
Remove WordPressKit from CocoaPods dependencies

All the sources are now part of the WordPressAuthenticator target in
WooCommerce.

mokagio committed:
Get WooCommerce to build successfully with new WordPressAuth. setup

mokagio committed:
Get WooCommerce target's unit tests to build with the new setup

mokagio committed:
Delete Jetpack Scan sources

mokagio committed:
Expect WordPressAuthenticatorTests failure in non-strict mode

mokagio committed:
Delete Jetpack Social

mokagio committed:
Delete Jetpack AI / Assistant sources

mokagio committed:
fixup! Delete Jetpack Social

mokagio committed:
Delete Jetpack Backup

mokagio committed:
Delete Jetpack Service sources

mokagio committed:
Remove `ActivityServiceRemote_ApiVersion1_0`

mokagio committed:
Delete `JetpackRestoreTypes`

mokagio committed:
Delete `ActivityServiceRemote`

mokagio committed:
Delete unused `plugin-state-*` JSONs

mokagio committed:
Delete plugin directory logic

mokagio committed:
Remove Jetpack capabilities

mokagio committed:
Delete `MockPluginDirectoryEntryProvider`

mokagio committed:
Delete `BlogJetpackSettingsServiceRemote` and model types

mokagio committed:
Remove unused `plugin-directory-*` JSONs

mokagio committed:
Delete `JetpackPluginManagementClient`

mokagio committed:
Delete Jetpack proxy service

mokagio committed:
Remove some unused code

mokagio committed:
Delete duplicated imports in WordPressAuthenticator header

mokagio committed:
Consolidate logging in WordPressAuth + WordPressKit

mokagio committed:
Remove more unused code

mokagio committed:
Delete no-longer necessary tests for WordPressKit logging

mokagio committed:
Port changes from WordPressAuthenticator Woo Brand Update PR to project

When https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/867
was merged, I had already started working on folding
WordPressAuthenticator and didn't sync with the team about it.

This commit picks all the changes from that PR and copies them in one go
into the local version of the WordPressAuthenticator codebase.

mokagio committed:
Remove shared Workspace `xcconfig` because it was not shared

It turned out that every project made a copy of that file, which
defeated the whole purpose.

Apparently, we cannot have files shared at the workspace level.

It might still be possible to achieve the same result in the future, but
that would require every project being set up with base configuration files
and referencing the shared file from there, without adding the shared file
itself to the projects. But to do that, we'll need each project set up
that way, which is out of scope for the work I'm currently doing of
folding the WordPressAuthenticator and WordPressKit libraries into
WooCommerce iOS.

mokagio committed:
Set iOS deployment target to 16.0 manually in projects

16.0 is the same as WooCommerce iOS uses

mokagio committed:
Delete MenusService and related models

mokagio committed:
Delete MediaServiceRemote and related models

mokagio committed:
Deleted `EditorServiceRemote`

mokagio committed:
Remove `RemoteBlockEditorSettings` and related models

mokagio committed:
Delete Reader services and models

mokagio committed:
Remove Revisions services and models

mokagio committed:
Remove Publicize services and models

mokagio committed:
Delete Stats services and models

mokagio committed:
Remove Theme services and model

mokagio committed:
Delete Timezones service and model

mokagio committed:
Remove countries and states services and models

mokagio committed:
Remove Comment services and models

mokagio committed:
Delete blogging prompt models and services

mokagio committed:
Remove Gravatar model and service

mokagio committed:
Remove page layout service and models

mokagio committed:
Remove unused `RemoteInviteLink`

mokagio committed:
Remove RemoteDomain service, model, and downstream usages

mokagio committed:
Delete unused `RemotePlan_ApiVersion1_3`

mokagio committed:
Delete `RemoteProfile` model and related services

mokagio committed:
Delete `RemoteShareAppContent` model and related service

mokagio committed:
Delete unused JSON and XML fixtures

mokagio committed:
Delete `RemoteSiteDesign` model and related service

mokagio committed:
Delete `SiteManagementServiceRemote`

mokagio committed:
Delete `ProductServiceRemote`

mokagio committed:
Delete `SelfHostedPluginManagementClient`

mokagio committed:
Delete `RemoteWpcomPlan`

mokagio committed:
Delete `SharingServiceRemote`

mokagio committed:
Delete `ServiceRequest` and related service

mokagio committed:
Remove `RemoteUser+Likes`

mokagio committed:
Delete `RemotePostType`

mokagio commented on Podfile:
```diff
@@ -37,7 +37,11 @@ def tracks
 
 def wordpress_shared
-  pod 'WordPressShared', '~> 2.1'
+  pod 'WordPressShared', '~> 2.1-beta'
```
Comes from WordPressKit https://github.com/wordpress-mobile/WordPressKit-iOS/blob/40a984ddc1f6775052f81b82565c1fb01d8198c5/WordPressKit.podspec#L30-L32

mokagio commented on config/Common.xcconfig:
```diff
@@ -1,4 +1,4 @@
-IPHONEOS_DEPLOYMENT_TARGET = 15.0
+IPHONEOS_DEPLOYMENT_TARGET = 16.0
```
Already set in the project but not update here and ignored because some target overrode it instead of inheriting (by deleting the entry at the target level)

mokagio committed:
Remove no-longer-needed SVProgressHUD import in main target

hichamboushaba commented:
Hey @mokagio, when can we expect this PR to be ready for review and merged? We have a project that touches on this code (pe5sF9-3mV-p2), and we are delaying it until this PR is merged to avoid any more unwanted conflicts.

mokagio committed:
Revert "Avoid archive error when building without fetching code signing"

This reverts commit d89e961bf37dd4e4e768b2708013693e1a78c5a1.

mokagio commented:
@hichamboushaba sorry you had to write that. Bad timing on my end... I had planned and started to work on this today p1737328507842779-slack, but something else came up, p1737344967048199-slack, so I only got to this now.

Anyway, this is now ready to review and I hope it'll get looked at soon. Feel free to leave your review, too, much appreciated.

wzieba commented:
hi @mokagio 👋 have you considered using `git filter-repo` to preserve the git history of the `WordPressAuthenticator` files? We've been using this strategy for a few Android projects now (Media, FluxC, Login) with success.

mokagio commented:
@wzieba 

> hi @mokagio 👋 have you considered using `git filter-repo` to preserve the git history of the WordPressAuthenticator files?

I didn't know about it before you mentioned it. Had a go at it in #15000

mokagio commented:
Closing in favor of #15000 as the only feedback I got on the team on preserving vs ignoring the library's history was positive towards the former.
